skimage/transform/_warps.py

```diff
@@ -831,8 +834,9 @@ def warp(image, inverse_map, map_args={}, output_shape=None, order=1,
             matrix = inverse_map.params
 
         elif (hasattr(inverse_map, '__name__') and
-              inverse_map.__name__ == 'inverse' and
+              inverse_map.__name__ == 'inverse_map' and
               get_bound_method_class(inverse_map) in HOMOGRAPHY_TRANSFORMS):
+            # TODO: do we still need this introspection hack?
             # inverse_map is the inverse of a homography
             matrix = np.linalg.inv(six.get_method_self(inverse_map).params)
```

Comment:
In Python 3, to get the bound method's instance, use `method.__self__`. Additionally, call `inverse_map.__self__.inverse` here instead. Don't forget to remove the `six` import up top!

Reply:
Ok, I think I get it. Warping is trying to get at the mathematical representation of the transform and not the functional one to speed things up.

This code wouldn't even work with PeicewiseAffine, which is the last case I was considering.